src/App.tsx

```diff
@@ -1,72 +1,97 @@
-import React from 'react';
+import React, { useState, useMemo } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
-export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+type Props = {
+  onDebounce?: (delay: number) => void;
+  onSelected?: (person: Person[] | null) => void;
+  debounceDelay?: number;
+};
+
+export const App: React.FC<Props> = ({
+  onDebounce,
+  onSelected,
+  debounceDelay = 300,
+}) => {
+  const [query, setQuery] = useState('');
+  const [people] = useState(peopleFromServer);
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [isListVisible, setIsListVisible] = useState(true);
+
+  const debouncedSearch = useMemo(() => {
+    const timeoutId = setTimeout(() => {
+      onDebounce?.(debounceDelay);
+    }, debounceDelay);
+
+    return () => clearTimeout(timeoutId);
+  }, [debounceDelay, onDebounce]);
```

Comment:
The `debouncedSearch` function should return a cleanup function that cancels the timeout when the component unmounts or when the dependencies change. Consider using `useEffect` instead of `useMemo` for this purpose.